src/unix/stdpaths.cpp

```diff
@@ -167,6 +176,37 @@ wxString wxStandardPaths::GetExecutablePath() const
 
     if ( !exeStr.empty() )
         return exeStr;
+#elif defined(__FreeBSD__)
+    struct procstat *ps = nullptr;
+    struct kinfo_proc *procs = nullptr;
+    char pathname[PATH_MAX];
+
+    do {
+        unsigned int n_proc;
+
+        ps = procstat_open_sysctl();
+        if (ps == nullptr)
+            break;
+
+        procs = procstat_getprocs(ps, KERN_PROC_PID, getpid(), &n_proc);
+        if (procs == nullptr)
+            break;
+
+        if (n_proc != 1)
+            break;
+
+        procstat_getpathname(ps, procs, pathname, sizeof(pathname));
+    } while (false);
```

Comment:
The `do/while(false)` hack is rather ugly and unnecessary in C++ code. It would be nicer to rewrite this without, e.g.

```cpp
auto* const ps = procstat_open_sysctl();
if (!ps)
    return wxStandardPathsBase::GetExecutablePath();
wxON_BLOCK_EXIT1(procstat_close, ps);

auto* const procs = procstat_getprocs();
if (!procs)
    return wxStandardPathsBase::GetExecutablePath();
wxON_BLOCK_EXIT2(procstat_freeprocs, ps, procs);

if (procstat_getpathname(ps, procs, pathname, sizeof(pathname)) != 0)
    return wxStandardPathsBase::GetExecutablePath();

return pathname;
```